src/hmac.rs

```diff
@@ -245,11 +268,91 @@ impl Key {
     pub fn digest_algorithm(&self) -> &'static digest::Algorithm {
         self.ctx_prototype.inner.algorithm()
     }
+
+    /// Calculates the HMAC of `data` using the key in one step.
+    ///
+    /// Use [`sign_parts`] if the data is not in a contiguous memory region. In
+    /// more involved cases, use [`Context`] to calculate HMACs where the input
+    /// is not available at once.
+    ///
+    /// It is generally not safe to implement HMAC verification by comparing the
+    /// return value of `sign` to a tag. Use [`verify`] for verification instead.
+    ///
+    /// [`sign_parts`]: #method.sign_parts
+    /// [`verify`]: #method.verify
+    /// [`Context`]: ./struct.Context.html
+    pub fn sign(&self, data: &[u8]) -> Tag {
+        let mut ctx = Context::with_key(self);
```

Comment:
This pattern always feels a bit dirty, using `Option` just for its `IntoIterator` but as long as `[_; 1]` can't provide it yes.

Reply:
Ya, Ideally we could do `[data]` instead of `Some(data)`, but unfortunately https://github.com/rust-lang/rust/issues/25725 is still not fixed, so `Some` is the best we have.